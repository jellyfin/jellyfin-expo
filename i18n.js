/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
import * as Localization from 'expo-localization';
import i18next from 'i18next';
import { initReactI18next } from 'react-i18next';
import { I18nManager } from 'react-native';

import ar from './langs/ar.json';
import be from './langs/be.json';
import bg from './langs/bg.json';
import ca from './langs/ca.json';
import cs from './langs/cs.json';
import cy from './langs/cy.json';
import da from './langs/da.json';
import de from './langs/de.json';
import el from './langs/el.json';
import en from './langs/en.json';
import eo from './langs/eo.json';
import es from './langs/es.json';
import es_419 from './langs/es_419.json';
import es_AR from './langs/es_AR.json';
import et from './langs/et.json';
import fa from './langs/fa.json';
import fi from './langs/fi.json';
import fil from './langs/fil.json';
import fr from './langs/fr.json';
import ga from './langs/ga.json';
import he from './langs/he.json';
import hr from './langs/hr.json';
import hu from './langs/hu.json';
import hy from './langs/hy.json';
import id from './langs/id.json';
import is from './langs/is.json';
import it from './langs/it.json';
import ja from './langs/ja.json';
import kk from './langs/kk.json';
import ko from './langs/ko.json';
import lt from './langs/lt.json';
import lv from './langs/lv.json';
import mn from './langs/mn.json';
import ms from './langs/ms.json';
import mt from './langs/mt.json';
import nb_NO from './langs/nb_NO.json';
import nds from './langs/nds.json';
import nl from './langs/nl.json';
import pl from './langs/pl.json';
import pt from './langs/pt.json';
import pt_BR from './langs/pt_BR.json';
import pt_PT from './langs/pt_PT.json';
import ro from './langs/ro.json';
import ru from './langs/ru.json';
import sk from './langs/sk.json';
import sl from './langs/sl.json';
import sq from './langs/sq.json';
import sr from './langs/sr.json';
import sv from './langs/sv.json';
import ta from './langs/ta.json';
import tr from './langs/tr.json';
import uk from './langs/uk.json';
import vi from './langs/vi.json';
import zh_Hans from './langs/zh_Hans.json';
import zh_Hant from './langs/zh_Hant.json';

export const resources = {
	en: { translation: en },
	ar: { translation: ar },
	be: { translation: be },
	bg: { translation: bg },
	ca: { translation: ca },
	cs: { translation: cs },
	cy: { translation: cy },
	da: { translation: da },
	de: { translation: de },
	el: { translation: el },
	eo: { translation: eo },
	es: { translation: es },
	'es-419': { translation: es_419 },
	'es-AR': { translation: es_AR },
	et: { translation: et },
	fa: { translation: fa },
	fi: { translation: fi },
	fil: { translation: fil },
	fr: { translation: fr },
	ga: { translation: ga },
	he: { translation: he },
	hr: { translation: hr },
	hu: { translation: hu },
	hy: { translation: hy },
	id: { translation: id },
	is: { translation: is },
	it: { translation: it },
	ja: { translation: ja },
	kk: { translation: kk },
	ko: { translation: ko },
	lt: { translation: lt },
	lv: { translation: lv },
	mn: { translation: mn },
	ms: { translation: ms },
	mt: { translation: mt },
	'nb-NO': { translation: nb_NO },
	nds: { translation: nds },
	nl: { translation: nl },
	pl: { translation: pl },
	pt: { translation: pt },
	'pt-BR': { translation: pt_BR },
	'pt-PT': { translation: pt_PT },
	ro: { translation: ro },
	ru: { translation: ru },
	sk: { translation: sk },
	sl: { translation: sl },
	sq: { translation: sq },
	sr: { translation: sr },
	sv: { translation: sv },
	ta: { translation: ta },
	tr: { translation: tr },
	uk: { translation: uk },
	vi: { translation: vi },
	'zh-Hans': { translation: zh_Hans },
	'zh-Hant': { translation: zh_Hant }
};

// Ensure RTL layout is enabled for RTL locales
I18nManager.forceRTL(Localization.isRTL);

i18next
	.use(initReactI18next)
	.init({
		// debug: true,
		fallbackLng: 'en',
		// Use the fallback language for empty strings
		returnEmptyString: false,
		lng: Localization.locale,
		interpolation: {
			escapeValue: false
		},
		resources
	});

// Export i18next instance for use in tests
export default i18next;
