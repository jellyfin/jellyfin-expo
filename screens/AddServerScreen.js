/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
import { useNavigation } from '@react-navigation/native';
import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { Image, KeyboardAvoidingView, Platform, StyleSheet, View } from 'react-native';
import { Icon, Text, ThemeContext } from 'react-native-elements';
import { SafeAreaView } from 'react-native-safe-area-context';

import ServerInput from '../components/ServerInput';
import Screens from '../constants/Screens';
import { useStores } from '../hooks/useStores';
import { getIconName } from '../utils/Icons';

const AddServerScreen = () => {
	const navigation = useNavigation();
	const { t } = useTranslation();
	const { settingStore } = useStores();
	const { theme } = useContext(ThemeContext);

	return (
		<KeyboardAvoidingView
			style={{
				...styles.screen,
				backgroundColor: theme.colors.background
			}}
			behavior={Platform.OS == 'ios' ? 'padding' : 'height'}
		>
			<SafeAreaView
				style={styles.container}
				edges={[ 'right', 'bottom', 'left' ]}
			>
				<View style={styles.logoContainer}>
					<Image
						style={styles.logoImage}
						source={
							settingStore.getTheme().dark ?
								require('../assets/images/logo-dark.png') :
								require('../assets/images/logo-light.png')
						}
						fadeDuration={0} // we need to adjust Android devices (https://facebook.github.io/react-native/docs/image#fadeduration) fadeDuration prop to `0` as it's default value is `300`
					/>
				</View>
				<View>
					<ServerInput
						label={
							<View style={styles.labelContainer}>
								<Text
									style={{
										...styles.label,
										color: theme.colors.grey1
									}}
								>
									{t('addServer.address')}
								</Text>
								<Icon
									type='ionicon'
									name={getIconName('help-circle')}
									containerStyle={styles.icon}
									color={theme.colors.black}
									onPress={() => {
										navigation.navigate(Screens.ServerHelpScreen);
									}}
								/>
							</View>
						}
						placeholder='https://jellyfin.org'
					/>
				</View>
			</SafeAreaView>
		</KeyboardAvoidingView>
	);
};

const styles = StyleSheet.create({
	screen: {
		flex: 1
	},
	container: {
		flex: 1,
		justifyContent: 'space-evenly'
	},
	logoContainer: {
		alignSelf: 'center',
		paddingVertical: 10,
		height: '40%',
		maxHeight: 151,
		maxWidth: '90%'
	},
	logoImage: {
		flex: 1,
		resizeMode: 'contain',
		maxWidth: '100%'
	},
	labelContainer: {
		flexDirection: 'row',
		alignItems: 'flex-end'
	},
	label: {
		fontSize: 16,
		fontWeight: 'bold'
	},
	icon: {
		paddingHorizontal: 10
	}
});

export default AddServerScreen;
