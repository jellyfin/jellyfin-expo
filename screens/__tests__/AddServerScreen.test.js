/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

import { NavigationContainer } from '@react-navigation/native';
import { act, render } from '@testing-library/react-native';
import React from 'react';

import '../../i18n';
import AddServerScreen from '../AddServerScreen';

describe('AddServerScreen', () => {
	it('should render correctly', () => {
		const { toJSON, unmount } = render(
			<NavigationContainer>
				<AddServerScreen />
			</NavigationContainer>
		);

		expect(toJSON()).toMatchSnapshot();
		act(unmount);
	});
});
