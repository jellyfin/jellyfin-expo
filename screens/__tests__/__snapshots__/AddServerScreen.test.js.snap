// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddServerScreen should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "backgroundColor": undefined,
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "off",
      }
    }
    style={
      {
        "flex": 1,
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      style={
        {
          "alignSelf": "center",
          "height": "40%",
          "maxHeight": 151,
          "maxWidth": "90%",
          "paddingVertical": 10,
        }
      }
    >
      <Image
        fadeDuration={0}
        source={
          {
            "testUri": "../../../assets/images/logo-dark.png",
          }
        }
        style={
          {
            "flex": 1,
            "maxWidth": "100%",
            "resizeMode": "contain",
          }
        }
      />
    </View>
    <View>
      <View
        style={
          {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#43484d",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Server Address
          </Text>
          <View
            style={
              {
                "overflow": "hidden",
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#303337",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 3,
              "flexDirection": "row",
              "marginBottom": 12,
              "marginTop": 8,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
          <TextInput
            autoCapitalize="none"
            autoCompleteType="off"
            autoCorrect={false}
            autoFocus={true}
            editable={true}
            keyboardType="url"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="https://jellyfin.org"
            placeholderTextColor="#86939e"
            returnKeyType="go"
            selectionColor="#2089dc"
            style={
              {
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="server-input"
            textContentType="URL"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
